src/main/java/net/minestom/server/lock/AcquirableCollection.java

```diff
@@ -0,0 +1,24 @@
+package net.minestom.server.lock;
+
+import net.minestom.server.utils.collection.CollectionView;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Collection;
+import java.util.function.Consumer;
+
+public class AcquirableCollection<E extends LockedElement> extends CollectionView<E, Acquirable<E>> {
```

Comment:
Part of me thinks we should just be using Iterable here. We don't need the benefits afforded by the collection interface and it's much simpler.